### PR TITLE
allow broadcasting over horizontal spaces

### LINKED ISSUE #1331: Broadcast 2d shells over 3d sphere
Needed for ClimaAtmos without `bycolumn`.

Currently broken here:
https://github.com/CliMA/ClimaCore.jl/blob/9157f35a71147bf2ee4dfaad880ff70781340782/test/Fields/field.jl#L432-L444

### PR DESCRIPTION
**Pull request by simonbyrne**
Fixes #1331.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Review by charleskawczynski — APPROVED** [review]
This looks great! I'm realizing now that the test I included exercises the correct broadcast parts, but there's no test. Also: can you confirm that this is equivalent behavior to the point-domain broadcasting?

I.e., a single point broadcasts over the entire column, or entire domain

**Comment by simonbyrne** [comment]
i don't have a good test for that yet...

**Comment by charleskawczynski** [comment]
This PR should basically allow us to do explicit held suarez and aqua planet runs (right now we are limited to baroclinic wave, which excludes vertical diffusion).

File: test/Fields/field.jl [hunk 1]
```diff
@@ -430,21 +430,13 @@ end
...
-            domain_surface_bc!(Y.x, ᶜz_surf, ᶜx_surf)
-        end
+        domain_surface_bc!(Y.x, ᶜz_surf, ᶜx_surf)
+
         # Skip spaces incompatible with Fields.bycolumn:
         TU.bycolumnable(space) || continue
-        column_surface_bc!(Y.x, ᶜz_surf, ᶜx_surf)
+        Yc = TU.FieldFromNamedTuple(space, (; x = FT(1)))
+        column_surface_bc!(Yc.x, ᶜz_surf, ᶜx_surf)
+        @test Y.x == Yc.x
```
Comment (simonbyrne) [inline test/Fields/field.jl:439]: Is this reasonable? Not sure how best to test this.

Comment (charleskawczynski) [reply test/Fields/field.jl:439]: I think so

**Pull request merged**